core/src/main/java/com/datastax/oss/driver/internal/core/ssl/SniSslEngineFactory.java

```diff
@@ -44,17 +46,40 @@ public SSLEngine newSslEngine(@NonNull EndPoint remoteEndpoint) {
               "Configuration error: can only use %s with SNI end points",
               this.getClass().getSimpleName()));
     }
-    SSLEngine engine;
     SniEndPoint sniEndPoint = (SniEndPoint) remoteEndpoint;
+    InetSocketAddress address = sniEndPoint.resolve();
+    String sniServerName = sniEndPoint.getServerName();
 
-    engine = sslContext.createSSLEngine();
+    // When hostname verification is enabled (with setEndpointIdentificationAlgorithm), the SSL
```

Comment:
The thing that worries me is that we don't have any test that captures that. If somewhen in a future this method 
https://github.com/JetBrains/jdk8u_jdk/blob/master/src/share/classes/sun/security/ssl/X509TrustManagerImpl.java#L401-L445 will change we may not notice that port is taken into account. I think we should cover at least happy path